src/main/java/com/oltpbenchmark/api/Procedure.java

```diff
@@ -51,11 +52,12 @@ protected Procedure() {
      * the constructor, otherwise we can't get access to all of our SQLStmts.
      *
      * @param <T>
+     * @param workConf Workload configuration
      * @return
      */
     @SuppressWarnings("unchecked")
-    protected final <T extends Procedure> T initialize(DatabaseType dbType) {
-        this.dbType = dbType;
+    protected final <T extends Procedure> T initialize(WorkloadConfiguration workConf) {
+        this.workConf = workConf;
```

Comment:
I don't think we should do this. It makes it more brittle since `WorkloadConfiguration` is basically a dumping ground for everything. Let's revert this and only pass in `DatabaseType`.

Reply:
Done.